gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java

```diff
@@ -781,6 +818,8 @@ public void onContainersAllocated(List<Container> containers) {
               instanceName = null;
             }
           }
+          helixTagAllocatedContainerCount.put(containerHelixTag,
```

Comment:
Is this operation thread-safe? What happens if onContainersAllocated() is called from multiple callback threads?

Reply:
Yes, it need to be thread safe, so I put it within the synchronized block.